### PR TITLE
Add declarative config support for RuleBasedRoutingSampler

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1060.

### CODE REVIEW

**Comment by jack-berg** [comment]
Also, I'm happy to be added as a component owner for `:samplers`.

**Comment by trask** [comment]
> Also, I'm happy to be added as a component owner for `:samplers`.

that would be great, can you update in this PR?

File: samplers/README.md [hunk 1]
```diff
@@ -1,7 +1,65 @@
...
+
+The following samplers support [declarative configuration](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/configuration#declarative-configuration):
+
+* `RuleBasedRoutingSampler`
+
+To use:
+
+* Add a dependency on `io.opentelemetry:opentelemetry-sdk-extension-incubator:<version>`
+* Follow the [instructions](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/incubator/README.md#file-configuration) to configure OpenTelemetry with declarative configuration.
+* Configure the `.tracer_provider.sampler` to include the `rule_based_routing` sampler.
```
Comment (jack-berg) [inline samplers/README.md:13]: TODO: add usage instructions for otel java agent

Comment (jack-berg) [reply samplers/README.md:13]: Doesn't work with the agent yet:
- https://github.com/open-telemetry/opentelemetry-java/pull/6725
- https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12265

For now, add a note in the README that agent support is WIP.

**Pull request merged**